### PR TITLE
Take more care in handling statefulset pods in mongodb test

### LINKED ISSUE #12588: mongodb petsets extended tests: dns record of replica cannot be resolved
Seen here: https://ci.openshift.redhat.com/jenkins/job/origin_extended_image_tests/825/testReport/junit/(root)/Extended/_image_ecosystem__mongodb__Slow__openshift_mongodb_replication__with_petset__creating_from_a_template_should_process_and_create_the__https___raw_githubusercontent_com_sclorg_mongodb_container_master_examples_petset_mongodb_petset_persistent_yaml__template/

The test fails on reading record on second replica `mongodb-replicaset-1`:
```
Jan 20 04:39:16.600: INFO: Running 'oc exec --config=/tmp/extended-test-mongodb-petset-replica-jn4bx-blvrl-user.kubeconfig --namespace=extended-test-mongodb-petset-replica-jn4bx-blvrl mongodb-replicaset-1 -- bash -c mongo --quiet "$MONGODB_DATABASE" --username "$MONGODB_USER" --password "$MONGODB_PASSWORD" --eval 'rs.slaveOk(); printjson(db.test.find({}, {_id: 0}).toArray())''
Jan 20 04:39:16.911: INFO: Error running &{/data/src/github.com/openshift/origin/_output/local/bin/linux/amd64/oc [oc exec --config=/tmp/extended-test-mongodb-petset-replica-jn4bx-blvrl-user.kubeconfig --namespace=extended-test-mongodb-petset-replica-jn4bx-blvrl mongodb-replicaset-1 -- bash -c mongo --quiet "$MONGODB_DATABASE" --username "$MONGODB_USER" --password "$MONGODB_PASSWORD" --eval 'rs.slaveOk(); printjson(db.test.find({}, {_id: 0}).toArray())'] []   
2017-01-20T09:39:16.902+0000 E QUERY    [thread1] Error: Authentication failed. :
DB.prototype._authOrThrow@src/mongo/shell/db.js:1441:20
@(auth):6:1
@(auth):1:2

exception: login failed
```

This, in turn, seems to be caused by the first replica, `mongodb-replicaset-0`, not being able to resolve `mongodb-replicaset-1`'s hostname:
```
2017-01-20T09:39:02.354971000Z 2017-01-20T09:39:02.354+0000 W NETWORK  [conn7] getaddrinfo("mongodb-replicaset-1.mongodb-replicaset.extended-test-mongodb-petset-replica-jn4bx-blvrl.svc.cluster.local") failed: Name or service not known
2017-01-20T09:39:02.356061000Z 2017-01-20T09:39:02.355+0000 I NETWORK  [conn7] getaddrinfo("mongodb-replicaset-1.mongodb-replicaset.extended-test-mongodb-petset-replica-jn4bx-blvrl.svc.cluster.local") failed: Name or service not known
2017-01-20T09:39:02.356295000Z 2017-01-20T09:39:02.355+0000 I REPL     [conn7] replSetReconfig config object with 2 members parses ok
2017-01-20T09:39:02.357005000Z 2017-01-20T09:39:02.356+0000 I ASIO     [NetworkInterfaceASIO-Replication-0] Connecting to mongodb-replicaset-1.mongodb-replicaset.extended-test-mongodb-petset-replica-jn4bx-blvrl.svc.cluster.local:27017
2017-01-20T09:39:02.358586000Z 2017-01-20T09:39:02.358+0000 I ASIO     [NetworkInterfaceASIO-Replication-0] Failed to connect to mongodb-replicaset-1.mongodb-replicaset.extended-test-mongodb-petset-replica-jn4bx-blvrl.svc.cluster.local:27017 - HostUnreachable: HostUnreachable
2017-01-20T09:39:02.358807000Z 2017-01-20T09:39:02.358+0000 W REPL     [ReplicationExecutor] Failed to complete heartbeat request to mongodb-replicaset-1.mongodb-replicaset.extended-test-mongodb-petset-replica-jn4bx-blvrl.svc.cluster.local:27017; HostUnreachable: HostUnreachable
```
Can't seem to reproduce this locally.

### PR DESCRIPTION
**Pull request by jim-minter**
The majority of the work was done in https://github.com/sclorg/mongodb-container/pull/227 .  I don't think there's a dependency between the two PRs in terms of which one merges first.

Fixes #12588

### CODE REVIEW

File: test/extended/image_ecosystem/mongodb_replica_statefulset.go [hunk 1]
```diff
@@ -94,11 +95,21 @@ var _ = g.Describe("[image_ecosystem][mongodb][Slow] openshift mongodb replicati
 			err = oc.Run("delete").Args("pods", "--all", "-n", oc.Namespace()).Execute()
 			o.Expect(err).ShouldNot(o.HaveOccurred())
 
-			g.By("waiting for all pods to reach running status")
+			g.By("waiting for all pods to be gracefully deleted")
 			podNames, err = exutil.WaitForPods(
 				oc.KubeClient().Core().Pods(oc.Namespace()),
 				exutil.ParseLabelsOrDie("name=mongodb-replicaset"),
-				exutil.CheckPodIsRunningFn,
+				func(pod kapi.Pod) bool { return pod.DeletionTimestamp != nil },
```
Comment (bparees) [inline test/extended/image_ecosystem/mongodb_replica_statefulset.go]: are we guaranteed to see the pod while it has a deletion timestamp? what if it "gracefully deletes" very quickly?

Comment (bparees) [inline test/extended/image_ecosystem/mongodb_replica_statefulset.go]: nm

**Comment by bparees** [comment]
[testextended][extended:core(mongodb replication)]

**Comment by bparees** [comment]
lgtm pending tests passing

**Comment by bparees** [comment]
[merge]

**Comment by bparees** [comment]
flake https://github.com/openshift/origin/issues/10663
flake https://github.com/openshift/origin/issues/12911
[merge]

**Comment by stevekuznetsov** [comment]
@openshift-bot, the [last build](https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_future/130/) failed from the following flakes:
 - [integration](https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_integration_future/128/consoleFull#6856043505762928fe4b031e1b25ced2a) failed on: https://github.com/openshift/origin/issues/12927

re[merge]

**Pull request merged**